roachtest: clean up command-line flags

RaduBerinde opened the pull request:
**This PR is only for the last commit. The rest are #111811**

---

#### roachtest: clean up command-line flags

The code around command-line flags is pretty messy: they're defined in
many places; the name and description of a flag are far away from
the variable; the variable names look like local variables and in many
cases it's not obvious we're accessing a global.

This commit moves all flags to a separate subpackage,
`roachtestflags`, making all uses of global flags obvious. We also add
a bit of infrastructure to allow defining all information about a flag
right next to where the variable is declared.

We also provide a `Changed()` function that determines if a flag value
was changed (without the caller having to use the Command or even the
flag name).

There should be no functional changes (just some cosmetic improvements
to the flag usage texts).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112078)
<!-- Reviewable:end -->

RaduBerinde commented:
This isn't urgent and I will be out next week in case there is any fallout; I will pick it back up when I come back.

RaduBerinde commented:
Updated, ready for review.

renatolabs reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Thanks for the refactor!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112078#-:-NiLiIi31l2Rvh7DSQNw:b-7yowd)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @DarrylWong and @herkolategan)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachtest: clean up command-line flags

The code around command-line flags is pretty messy: they're defined in
many places; the name and description of a flag are far away from
the variable; the variable names look like local variables and in many
cases it's not obvious we're accessing a global.

This commit moves all flags to a separate subpackage,
`roachtestflags`, making all uses of global flags obvious. We also add
a bit of infrastructure to allow defining all information about a flag
right next to where the variable is declared.

We also provide a `Changed()` function that determines if a flag value
was changed (without the caller having to use the Command or even the
flag name).

There should be no functional changes (just some cosmetic improvements
to the flag usage texts).

Epic: none
Release note: None
